master/internal/experiment_filter.go

```diff
@@ -361,9 +361,16 @@ func (e experimentFilter) toSQL(q *bun.SelectQuery,
 			case doesNotContain:
 				queryString = "? NOT ILIKE ?"
 				queryArgs = append(queryArgs, bun.Safe(col), fmt.Sprintf("%%%s%%", *e.Value))
-			case empty, notEmpty:
-				queryString = "? ?"
+			case empty:
+				queryString = "? ? OR ? ? ? OR ? ? ?"
+				queryArgs = append(queryArgs, bun.Safe(col), bun.Safe(oSQL))
+				queryArgs = append(queryArgs, bun.Safe(col), bun.Safe(equal), "")
```

Comment:
total nit, it looks like `equal` and `notEqual` are used to match on the input format, so using them in the output query is maybe unintended coupling
```suggestion
				queryArgs = append(queryArgs, bun.Safe(col), bun.Safe(equal), "")
```